python/ray/tests/test_autoscaler_aws.py

```diff
@@ -0,0 +1,316 @@
+from datetime import datetime
+
+import pytest
+import copy
+
+from unittest import mock
+
+from botocore.stub import ANY
+
+import ray.autoscaler.aws.config
+from ray.autoscaler.aws.config import key_pair
+
+# Override global constants used in AWS autoscaler config artifact names.
+# This helps ensure that any unmocked test doesn't alter non-test artifacts.
+ray.autoscaler.aws.config.RAY = \
+    "ray-autoscaler-aws-test"
+ray.autoscaler.aws.config.DEFAULT_RAY_INSTANCE_PROFILE = \
+    ray.autoscaler.aws.config.RAY + "-v1"
+ray.autoscaler.aws.config.DEFAULT_RAY_IAM_ROLE = \
+    ray.autoscaler.aws.config.RAY + "-v1"
+ray.autoscaler.aws.config.SECURITY_GROUP_TEMPLATE = \
+    ray.autoscaler.aws.config.RAY + "-{}"
+
+# Default IAM instance profile to expose to tests.
+DEFAULT_INSTANCE_PROFILE = {
+    "Arn": "arn:aws:iam::336924118301:instance-profile/ExampleInstanceProfile",
+    "CreateDate": datetime(2013, 6, 12, 23, 52, 2, 2),
+    "InstanceProfileId": "AID2MAB8DPLSRHEXAMPLE",
```

Comment:
nit: I think technically this needs to be an `AIPA` (c/o https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_identifiers.html#identifiers-unique-ids)

Reply:
Good call - updated. This could come in handy if we ever start doing more strict validation of mocked inputs/outputs.